src/web/demo.rkt

```diff
@@ -328,19 +328,58 @@
        (when (eof-object? formula)
          (raise-herbie-error "no formula specified"))
        (parse-test formula)
-       (define hash (sha1 (open-input-string formula-str)))
+       (define hash (sha1 (open-input-string (~s 'improve formula (get-seed)))))
        (body hash formula))]
     [_
      (response/error "Demo Error"
                      `(p "You didn't specify a formula (or you specified several). "
                          "Please " (a ([href ,go-back]) "go back") " and try again."))]))
 
+(define (improve-start-extract post-data)
+  (define form-str (hash-ref post-data 'formula))
+  (define form-seed (hash-ref post-data 'seed))
+  (list form-str form-seed))
+
+(define (improve-start-helper req body go-back)
```

Comment:
Why define this to take a callback? This just seems to make the code hard to read. Why not just raise an exception, and do the exception handling in a separate method if you must.

Reply:
Honestly wasn't sure how to extract and bind the two values well reusing the code that was there to avoid changing to many things at once. Not really sure how to do what you're saying though.